ext/node/polyfills/http2.ts

```diff
@@ -840,6 +840,11 @@ async function clientHttp2Request(
     reqHeaders,
   );
 
+  if (session.closed || session.destroyed) {
+    debugHttp2(">>> session closed during request promise");
+    throw new ERR_HTTP2_STREAM_CANCEL();
```

Comment:
I tried the below script with Node.js (v22.7) and it doesn't seem throwing in this situation (The below exits without showing an error):

```js
import http2 from "node:http2";
const url = "http://127.0.0.1:4246";
const connectPromise = Promise.withResolvers();
const client = http2.connect(url, {}, () => {
  connectPromise.resolve();
})

await connectPromise.promise

// leave a request open to prevent immediate destroy
const req = client.request();
req.on("data", () => {});
req.on("error", (err) => console.error(err));
const reqClosePromise = Promise.withResolvers();
req.on("close", () => { reqClosePromise.resolve() });

client.close();

await reqClosePromise.promise
```

Can you share the reasoning why throwing `ERR_HTTP2_STREAM_CANCEL` here?

Reply:
All open and pending streams are destroyed with `ERR_HTTP2_STREAM_CANCEL` here:

https://github.com/denoland/deno/blob/7e68cce8159d55fd597f0da3e00f794200b9928f/ext/node/polyfills/http2.ts#L410

I initially tried to just catch the error and wait for that logic to kick in, but I got errors running tests. State machine stopped but there are still outstanding promises.

Happy to change it, what would you recommend?

Reply:
Looked into this more, and it looked like the root cause of this is that we don't implement GOAWAY frame and streams can't be closed/destroyed normally.

https://github.com/denoland/deno/blob/97d1635343dc6e93c8dcf4b116922de5b9c57af3/ext/node/polyfills/http2.ts#L316-L317

I now think this implementation is a good compromise at this point.